### PR TITLE
add a setting to require TLS

### PR DESCRIPTION
**Pull request by drakkan**
TLSMode is an int since in future we could add other options, for
example require TLS for control connection but not for data connection.

@fclairamb thank you for giving me push permissions! I prefer a review anyway instead of pushing directly my changes

### CODE REVIEW

File: driver.go [hunk 1]
```diff
@@ -157,4 +157,9 @@ type Settings struct {
 	DisableMLST              bool             // Disable MLST support
 	DisableMFMT              bool             // Disable MFMT support (modify file mtime)
 	Banner                   string           // Banner to use in server status response
+	// 0 means accept both cleartext and encrypted sessions
+	// 1 means TLS in required for both control and data connection
+	// Do not enable this blindly, please check that a proper TLS config
+	// is in place or no login will be allowed
+	TSLMode int
```
Comment (fclairamb) [inline driver.go]: why didn't you declare it as
```golang
ForceTLS bool
```

ie:
- As a boolean
- With a more explicit name

Comment (drakkan) [reply driver.go]: Hi, `ForceTLS` maybe is a better name, ProFTPD calls it `TLSRequired` but I'm not sure a boolean is a good choice. Please take a look [here](http://www.proftpd.org/docs/directives/linked/config_ref_TLSRequired.html
) for other possible choices we could add in future.

Comment (fclairamb) [reply driver.go]: I see.

I guess we can switch to `TLSRequired` with as an `int` and then later improve the code to use typed constants here (when we will support other TLS mode requirements).

Comment (drakkan) [reply driver.go]: Fixed, thank you

**Pull request merged**